src/modules/constructRelationalGraph.js

```diff
@@ -0,0 +1,252 @@
+// @flow
+import Doctor from '../resources/Doctor.json';
+import Patient from '../resources/Patient.json';
+import Department from '../resources/Department.json';
+
+import type {
+  JoinCondition,
+  SelectionCondition,
+  Expression,
+  Graph,
+} from './types';
+
+let canOptimize = true;
```

Comment:
Try to avoid making use of global variables. This can make following the logic rather confusing.